build.gradle

```diff
@@ -181,6 +181,41 @@ tasks.register("configureArtifactInfo") {
     project.ext.set("stackArtifactSuffix", qualifiedVersion)
 }
 
+tasks.register("markAliasDefinitions") {
+    description "Create an hashes aliases file from original aliases yml definition"
+    inputs.file("${projectDir}/logstash-core/src/main/resources/org/logstash/plugins/AliasRegistry.yml")
+    outputs.file("${projectDir}/plugin_aliases_hashed.yml")
+    doLast {
+        String aliases_defs = new File("${projectDir}/logstash-core/src/main/resources/org/logstash/plugins/AliasRegistry.yml").text
```

Comment:
here we read the text as-is to make the checksum, but elsewhere we chomp/strip it first. This will cause problems if we check in a file that _HAS_ a trailing newline.

Reply:
we `chomp` and `strip` only the checksum header (https://github.com/elastic/logstash/pull/12841/files#diff-bf2bef319b3c0c24389cdd5fa67a30ef14641ae1b5fc1180e371ced0a31c278eR36) in all the other locations where we touch the yaml text it's read to the end of file, without removing empty lines or so, I don't think that here we need to manipulate the provided text. The idea is to keep all that's defined in `AliasRegsitry.yml` but only adding 2 lines on the head